pkg/kwokctl/runtime/util.go

```diff
@@ -116,10 +119,30 @@ func applyComponentPatch(component *internalversion.Component, patch internalver
 
 	component.Volumes = append(component.Volumes, patch.ExtraVolumes...)
 	component.Envs = append(component.Envs, patch.ExtraEnvs...)
-
 	for _, a := range patch.ExtraArgs {
-		component.Args = append(component.Args, fmt.Sprintf("--%s=%s", a.Key, a.Value))
+		if a.Override {
+			component.Args = applyComponentArgsOverride(component.Args, a)
+		} else {
+			component.Args = append(component.Args, fmt.Sprintf("--%s=%s", a.Key, a.Value))
+		}
+	}
+}
+
+func applyComponentArgsOverride(args []string, a internalversion.ExtraArgs) []string {
```

Comment:
```suggestion
func applyComponentArgsOverride(ctx context.Context, args []string, a internalversion.ExtraArgs) []string {
```

Let's pass the context